django_lightweight_queue/backends/reliable_redis.py

```diff
@@ -101,6 +101,28 @@ def processed_job(self, queue, worker_number, job):
     def length(self, queue):
         return self.client.llen(self._key(queue))
 
+    def deduplicate(self, queue):
+        main_queue_key = self._key(queue)
+
+        original_size = self.client.llen(main_queue_key)
+
+        if not original_size:
+            return 0, 0
+
+        jobs = {}
+
+        for raw_data in self.client.lrange(main_queue_key, 0, -1):
+            job_id = Job.from_json(raw_data).identity_without_created()
+
+            jobs.setdefault(job_id, []).append(raw_data)
+
+        for raw_jobs in jobs.values():
+            # Leave the first one in the queue
+            for raw_data in raw_jobs[1:]:
+                self.client.lrem(raw_data)
```

Comment:
This won't work as you intend it to :).
1. We should use an OrderedDict to preserve the ordering.
2. We actually want the _last_ of each job. Jobs are pushed onto the head of the queue, and popped from the tail (there's docs on the class that talk about this - but that's pretty much all you need to know).

Reply:
I originally thought about using an `OrderedDict` but then realised that we don't need to. I'll add a comment explaining why.